client/knode/validator/states.go

```diff
@@ -75,10 +41,53 @@ func (val *validator) notLoggedInState() stateFn {
 		if !isVoter {
 			log.Crit("Invalid genesis - genesis validator needs to be registered as a voter", "address", val.walletAccount.Account().Address)
 		}
-
 		log.Info("Deposit is not necessary for a genesis validator (first block)")
+
+		return val.startValidating
 	}
 
+	return val.notLoggedInState
+}
+
+func (val *validator) notLoggedInState() stateFn {
+	chainHeadCh := make(chan core.ChainHeadEvent)
+	chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
+	defer chainHeadSub.Unsubscribe()
```

Comment:
@rgeraldes can you explain what this is doing?  this func only has a call so the defer unsubscribes right after

Reply:
chainHeadSub lets us know when there's a new chain head (block) > check the state again to verify if the user is a validator.

Reply:
but chainHeadSub is not in use anywhere. And it's closes the channel as soon at exits that func

Reply:
that's true - we can remove it for now.